source/includes/read/cursors.kt

```diff
@@ -0,0 +1,71 @@
+package org.example
+import com.mongodb.ConnectionString
+import com.mongodb.CursorType
+import com.mongodb.MongoClientSettings
+import com.mongodb.client.model.Filters.eq
+import com.mongodb.kotlin.client.MongoClient
+import org.bson.codecs.pojo.annotations.BsonId
+import org.bson.types.ObjectId
+import org.bson.Document
+
+// start-data-class
+data class Restaurant(
+    @BsonId
+    val id: ObjectId,
+    val name: String
+)
+// end-data-class
+
+fun main() {
+    val uri = "<connection string URI>"
+
+    val settings = MongoClientSettings.builder()
+        .applyConnectionString(ConnectionString(uri))
+        .retryWrites(true)
+        .build()
+
+    val mongoClient = MongoClient.create(settings)
+    val database = mongoClient.getDatabase("sample_restaurants")
+    val collection = database.getCollection<Restaurant>("restaurants")
+
+    // start-cursor-iterate
+    val results = collection.find()
+
+    results.forEach { result ->
+        println(result)
+    }
+    // end-cursor-iterate
+
+    // start-cursor-iterate-alternative
+    val result = collection.find()
+
+    results.cursor().use { cursor ->
```

Comment:
The variable name is `result`, while the variable name used by the code that follows is `results`, that would be a compilation error.

```suggestion
    val results = collection.find()

    results.cursor().use { cursor ->
```

Reply:
Whoops, will get a quick fix out!